File: Scripts/Room/ShopInventory.cs
using Roguelike.Data;
using System;
using System.Collections.Generic;

namespace Roguelike.Logic
{
    public class ShopInventory
    {
        public List<ShopItem<CardData>> CardsForSale { get; } = new List<ShopItem<CardData>>();
        public List<ShopItem<RelicData>> RelicsForSale { get; } = new List<ShopItem<RelicData>>();

        public ShopInventory(CardPool cardPool, RelicPool relicPool, Random rng)
        {
            for (int star = 1; star <= 5; star++)
            {
                var card = cardPool.GetRandomCardOfStar(star, rng);
                if (card != null && cardPool.BaseShopCosts.TryGetValue(star, out int baseCardCost))
                {
                    int price = rng.Next(baseCardCost, baseCardCost + 30);
                    CardsForSale.Add(new ShopItem<CardData>(card, price));
                }

                var relic = relicPool.GetRandomRelicOfStar(star, rng);
                if (relic != null && relicPool.BaseShopCosts.TryGetValue(star, out int baseRelicCost))
                {
                    int price = rng.Next(baseRelicCost, baseRelicCost + 50);
                    RelicsForSale.Add(new ShopItem<RelicData>(relic, price));
                }
            }
        }
    }
}

File: python_roguelike/core/rooms/shop_inventory.py
import random
from typing import List
from .shop_item import ShopItem
from ...data.card_data import CardData
from ...data.relic_data import RelicData
from ...data.pools.card_pool import CardPool
from ...data.pools.relic_pool import RelicPool


class ShopInventory:
    def __init__(self, card_pool: CardPool, relic_pool: RelicPool, rng: random.Random):
        self.cards_for_sale: List[ShopItem] = []
        self.relics_for_sale: List[ShopItem] = []

        for star in range(1, 6):
            card = card_pool.get_random_card_of_star(star, rng)
            if card is not None and star in card_pool.base_shop_costs:
                base_card_cost = card_pool.base_shop_costs[star]
                price = rng.randint(base_card_cost, base_card_cost + 29)
                self.cards_for_sale.append(ShopItem(item=card, price=price))

            relic = relic_pool.get_random_relic_of_star(star, rng)
            if relic is not None and star in relic_pool.base_shop_costs:
                base_relic_cost = relic_pool.base_shop_costs[star]
                price = rng.randint(base_relic_cost, base_relic_cost + 49)
                self.relics_for_sale.append(ShopItem(item=relic, price=price))
